### PR TITLE
CON-2642 canjump solution and test

### PR DESCRIPTION
**Pull request by MSilva95**
CON-2642 

 canjump solution and test

### CODE REVIEW

File: solutions/canjump.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package solutions
+
+func CanJump(steps ...int) bool {
```
Comment (nprimo) [inline solutions/canjump.go:3]: A possible implementation to solve what  got from the subject:
```go
func CanJump(in []uint) bool {
	if len(in) == 0 {
		return false
	}
        if len(in) == 1 {
           return true
        }

	toReach := len(in) - 1
	var maxReached uint
	for i := uint(0); i <= uint(toReach); i++ {
		if maxReached < i {
			return false
		}
		for j := in[i]; j > 0; j-- {
			maxReached = max(maxReached, j+i)
			if maxReached >= uint(toReach) {
				return true
			}
		}
	}

	return false
}
```

**Review by nprimo — CHANGES_REQUESTED** [review]

File: solutions/canjump.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (MSilva95) [reply solutions/canjump.go:3]: my bad this was the first version, now its a uint as suggested

File: tests/canjump_test/main.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+		args []int
+		want bool
+	}{
+		{args: []int{2, 3, 1, 1, 4}, want: true},
+		{args: []int{1, 1, 1, 1, 0}, want: true},
+		{args: []int{5, 4, 3, 2, 1, 0}, want: true},
+		{args: []int{0}, want: true},
+		{args: []int{5}, want: true},
+		{args: []int{}, want: false},
+		{args: []int{1, 2, 3, 0, 2}, want: false},
```
[the remark this answers is not in the document]
Comment (nprimo) [reply tests/canjump_test/main.go]: The issue (for what I understand in the subject) is that the number represent "the maximum number of steps" and not "the precise number of steps", therefore I still don't think this test and the below ones should be `false`.

File: tests/canjump_test/main.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+				"CanJump",
+				uintSliceToString(tc.args),
+				got,
+				tc.want,
+			)
+		}
+	}
+}
+
+func uintSliceToString(nums []uint) string {
```
Comment (nprimo) [inline tests/canjump_test/main.go:45]: This function can be removed in my opinion - the `$#v` format the array with hexadecimal notation (`0x01`,...), but the `$+v` format it in a more "human friendly" way :)

Comment (MSilva95) [reply tests/canjump_test/main.go]: done

**Review by nprimo — APPROVED** [review]
LGTM

**Pull request merged**